src/pages/Swap/LimitOrderBox/utils/hooks.ts

```diff
@@ -0,0 +1,115 @@
+    const getLimitOrderTransactions = async (chainId: ChainId, account: string) => {
+      const orders = await getOrders(account, allNetworks, chainId)
+      if (orders && orders.length > 0) {
+        //@ts-expect-error  chainId here wont have other networks since its filtered in upper scope
+        const appData = allNetworks ? Object.values(appDataHashes) : [appDataHashes[chainId]]
+        const filteredOrders = orders.filter(order => appData.includes(order.appData.toString()))
+        const formattedLimitOrders = filteredOrders.map(order => {
+          return {
+            ...order,
+            type: 'Limit Order' as const,
+            hash: order.uid,
+            network: getNetworkId(order.appData.toString()),
+            sellToken: {
+              value: order.sellAmount,
+              symbol: '',
+              tokenAddress: order.sellToken,
+              chainId: getNetworkId(order.appData.toString()),
+            },
+            buyToken: {
+              value: order.buyAmount,
+              symbol: '',
+              tokenAddress: order.buyToken,
+              chainId: getNetworkId(order.appData.toString()),
+            },
+            confirmedTime: order.validTo * 1000,
+            status: order.status,
+            cancelOrder: cancelOpenOrder(order.status, chainId, () => getLimitOrderTransactions(chainId, account)),
+          }
+        })
+        setOrders(formattedLimitOrders)
+      }
+    }
+    if (account != null && chainId !== undefined) {
+      getLimitOrderTransactions(chainId, account)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [account, chainId, allNetworks])
+
+  return orders
+}
+
+function cancelOpenOrder(status: OrderMetaData['status'], chainId: ChainId, updateLimitOrderTransactions: () => void) {
+  if (status === 'open') {
+    return async (uid: string, provider: Web3Provider) => {
+      const signer = provider.getSigner()
+      // Cancel an open order
+      const response = await deleteOpenOrders(chainId, uid, signer)
+      if (response === 'Cancelled') {
+        // Refetch orders
+        await updateLimitOrderTransactions()
+        return true
+      } else {
+        return false
+      }
+    }
+  }
+}
+
+function getNetworkId(appData: string) {
+  return Object.keys(appDataHashes)
+    .map(Number)
+    .find((key: keyof typeof appDataHashes) => appDataHashes[key] === appData) as number
+}
+
+async function getOrders(account: string, allNetworks: boolean, chainId: ChainId) {
+  if (allNetworks) {
+    const allOrders = await Promise.all([
+      getOwnerOrders(ChainId.GNOSIS, account),
+      getOwnerOrders(ChainId.MAINNET, account),
+    ]).catch(error => {
+      console.error({ error, message: 'All limit order fetch failed' })
+      return [[]]
+    })
+    return allOrders.flat()
+  }
+  if (chainId === ChainId.GNOSIS || chainId === ChainId.MAINNET) {
+    return await getOwnerOrders(chainId, account)
+  }
+  return [] as OrderMetaData[]
+}
```

Comment:
these are not hooks

Reply:
I think this is fine. They are only being used in this file so it's good they are next to where it's being used.